application-model/src/main/java/org/bonitasoft/engine/business/application/xml/ApplicationNode.java

```diff
@@ -175,21 +98,19 @@ public void addApplicationMenu(ApplicationMenuNode applicationMenu) {
     public boolean equals(Object o) {
         if (this == o)
             return true;
-        if (o == null || getClass() != o.getClass())
+        if (o == null || !super.equals(o))
             return false;
         ApplicationNode that = (ApplicationNode) o;
-        return Objects.equals(token, that.token) && Objects.equals(version, that.version)
-                && Objects.equals(displayName, that.displayName) && Objects.equals(description, that.description)
-                && Objects.equals(profile, that.profile) && Objects.equals(homePage, that.homePage)
-                && Objects.equals(state, that.state) && Objects.equals(layout, that.layout)
-                && Objects.equals(theme, that.theme) && Objects.equals(iconPath, that.iconPath)
+        return Objects.equals(homePage, that.homePage) && Objects.equals(layout, that.layout)
+                && Objects.equals(theme, that.theme)
                 && Objects.equals(applicationPages, that.applicationPages)
                 && Objects.equals(applicationMenus, that.applicationMenus);
     }
 
     @Override
     public int hashCode() {
-        return Objects.hash(token, version, displayName, description, profile, homePage, state, layout, theme, iconPath,
-                applicationPages, applicationMenus);
+        // keep same as previous hash code
```

Comment:
Actually you can call the super hashcode within Objects.hash

Reply:
Except it won't return the same hash value as before... I don't know whether the current hashcode has been used and stored somewhere... Do you ?

Reply:
Ok :thinking: A field might be missing then? I don't know either if it is used or not :shrug:

Reply:
No I mean
`java.util.Objects.hash("s1", "s2", "s3") != java.util.Objects.hash(java.util.Objects.hash("s1", "s2"), "s3")`

Reply:
It probably doesn't matter if the hashcode is not stored in a database, but I can't tell for sure...

Reply:
Ok good to know this about Objects.hash !